native/explorer/src/series.rs

```diff
@@ -161,6 +161,16 @@ pub fn s_slice(series: ExSeries, offset: i64, length: usize) -> Result<ExSeries,
     Ok(ExSeries::new(series.slice(offset, length)))
 }
 
+#[rustler::nif(schedule = "DirtyCpu")]
+pub fn s_format(list: Vec<ExSeries>) -> Result<ExSeries, ExplorerError> {
+    Ok(list
+        .into_iter()
+        .reduce(|acc, s2| {
+            ExSeries::new(acc.utf8().unwrap().concat(s2.utf8().unwrap()).into_series())
```

Comment:
I think we could avoid the additional allocations if we first map all the chunked arrays (the `series.utf8()` part), and then `concat` in the reduce, and after that we create one series in the end. The way you did we are creating intermediate `ExSeries` and `Series` that are going to be dropped for most of the operations.

Maybe it's just cheaper to go with the approach from `s_concat`, that uses a `for` loop. That way you can also rely on the `?` to return errors, instead of using `unwrap()`. WDYT?

Reply:
https://github.com/elixir-nx/explorer/pull/520/commits/8655649cbe8ec80e84c82d3397157a387f9c953a